tests/network-tests/src/fixtures/council/ElectCouncilFixture.ts

```diff
@@ -5,17 +5,34 @@ import { blake2AsHex } from '@polkadot/util-crypto'
 import { GenericAccountId } from '@polkadot/types'
 import { assert } from 'chai'
 import { createType, registry } from '@joystream/types'
+import { QueryNodeApi } from '../../QueryNodeApi'
+import { Api } from '../../Api'
 
 export class ElectCouncilFixture extends BaseQueryNodeFixture {
+  protected candidatesAddresses: string[] | undefined
+
+  public constructor(api: Api, query: QueryNodeApi, candidatesAddresses: string[] | undefined = undefined) {
+    super(api, query)
+    this.candidatesAddresses = candidatesAddresses
+  }
+
+  protected async createCandidates(): Promise<string[]> {
+    const { councilSize, minNumberOfExtraCandidates } = this.api.consts.council
+    const numberOfCandidates = councilSize.add(minNumberOfExtraCandidates)
+    const addresses = (await this.api.createKeyPairs(numberOfCandidates.toNumber())).map(({ key }) => key.address)
+    return addresses
+  }
+
   public async execute(): Promise<void> {
     const { api, query } = this
-    const { councilSize, minNumberOfExtraCandidates } = api.consts.council
-    const numberOfCandidates = councilSize.add(minNumberOfExtraCandidates).toNumber()
+
+    // get/create candidates member accounts
+    const candidatesMemberAccounts = this.candidatesAddresses ?? (await this.createCandidates())
```

Comment:
Great to have the custom candidate addresses. Instead of having the special case for size of council = 1 only for which `this.candidatesAddresses` was introduced, we might as well just have all addresses generated for council candidates have custom addresses, created in `function createCandidates()` above, with pattern like `//CouncilCandidate//N` where you need to store the counter N and increment for each new candidate. Or to avoid saving this 'global' counter, you could have `//CouncilCandidate//R//N` - use the election round as `R` and `N` can just repeat 0..n for each election attempt.

Reply:
I have followed the second approach